packages/blockchain-utils-linking/src/ethereum.ts

```diff
@@ -16,11 +16,23 @@ type EthProviderOpts = {
 
 const CHAIN_NAMESPACE = 'eip155'
 
+const chainIdCache = new WeakMap<any, number>()
+async function requestChainId(provider: any): Promise<number> {
+  let chainId = chainIdCache.get(provider)
+  if (!chainId) {
+    const chainIdHex = await safeSend(provider, 'eth_chainId', [])
+    chainId = parseInt(chainIdHex, 16)
+    chainIdCache.set(provider, chainId)
+  }
+  return chainId
+}
+
 /**
  *  AuthProvider which can be used for Ethereum providers with standard interface
  */
 export class EthereumAuthProvider implements AuthProvider {
   readonly isAuthProvider = true
+  private _accountId: AccountId | undefined
 
   constructor(
     private readonly provider: any,
```

Comment:
the EthereumAuthProvider maintains a single long-lived instance of the `provider`.  Given that, why keep a map of provider to chainid at all, instead of caching a single chainid in the EthereumAuthProvider instance itself?

Reply:
Follow-up: I see that there is an exported free function `createLink` that can be called without an instance of EthereumAuthProvider.  Do you know why this is?  The other blockchains in blockchain-utils-linking don't seem to have corresponding free functions - they all require creating the link to go through an instance of the provider type.

Can we remove the free-function `createLink` in favor of `EthereumAuthProvider.createLink`, and by so doing remove the need to use WeakMap and instead just cache the single chainid used by the single `provider` instance that `EthereumAuthProvider` uses?

Reply:
Not sure exactly how these APIs are used upstream, but I think we should try to avoid making breaking changes right now. I think there's an opportunity to revamp the AuthProvider API once we start relying more on CACAO for authentication, but for now I think we should keep apis the same.

Reply:
FWIW this function isn't part of the AuthProvider API, its specific to EthereumAuthProvider.  But anyway your point is still fair.  We can leave it for now, it just makes this caching code more involved than it needs to be but that's fine.